File: ibkr_show_frontend/src/types/trades.ts
import type { PaginationInfo } from './common'

export interface TradeItem {
  account_id: string
  trade_date: string | null
  date_time: string | null
  currency: string | null
  symbol: string | null
  description: string | null
  asset_class: string | null
  buy_sell: string | null
  quantity: number | null
  trade_price: number | null
  proceeds: number | null
  ib_commission: number | null
  net_cash: number | null
  fifo_pnl_realized: number | null
  exchange: string | null
  order_type: string | null
  transaction_id: string | null
  trade_id: string | null
}

export interface TradeListResponse {
  items: TradeItem[]
  pagination: PaginationInfo
}

export interface TradeSummaryResponse {
  trade_count: number
  buy_count: number
  sell_count: number
  total_commission: number
  total_realized_pnl: number
  total_proceeds: number
  symbols_count: number
}

File: ibkr_show_backend/app/schemas/trades.py
from pydantic import BaseModel

from app.schemas.common import PaginationInfo


class TradeItem(BaseModel):
    account_id: str
    trade_date: str | None = None
    date_time: str | None = None
    currency: str | None = None
    symbol: str | None = None
    description: str | None = None
    asset_class: str | None = None
    buy_sell: str | None = None
    quantity: float | None = None
    trade_price: float | None = None
    proceeds: float | None = None
    ib_commission: float | None = None
    net_cash: float | None = None
    fifo_pnl_realized: float | None = None
    exchange: str | None = None
    order_type: str | None = None
    transaction_id: str | None = None
    trade_id: str | None = None


class TradeListResponse(BaseModel):
    items: list[TradeItem]
    pagination: PaginationInfo


class TradeSummaryResponse(BaseModel):
    trade_count: int
    buy_count: int
    sell_count: int
    total_commission: float
    total_realized_pnl: float
    total_proceeds: float
    symbols_count: int
